packages/react-components/src/components/chart/useChartsLegend.tsx

```diff
@@ -0,0 +1,72 @@
+import { InternalGraphicComponentGroupOption } from './types';
+import React, { useEffect, useState } from 'react';
+import { TableProps } from '@cloudscape-design/components/table/interfaces';
+import { GraphicComponentTextOption } from 'echarts/types/src/component/graphic/GraphicModel';
+import { SeriesOption } from 'echarts';
+import { LineSeriesOption } from 'echarts/types/src/chart/line/LineSeries';
+
+const useChartsLegend = ({
+  graphic,
+  series,
+}: {
+  graphic: InternalGraphicComponentGroupOption[];
+  series: SeriesOption[];
+}) => {
+  const legendColumnDefinition = {
+    id: 'Legends',
+    header: <div className='base-chart-legend-col-header'>Legends</div>,
+    cell: (e: { lineColor: string; name: string }) => (
+      <div className='base-chart-legend-row-data-container'>
+        <div
+          className='base-chart-legend-row-line-ind'
+          style={{
+            backgroundColor: e.lineColor,
+          }}
+        ></div>
+        <div>{e.name}</div>
+      </div>
+    ),
+    isRowHeader: true,
+  };
+
+  const getTcHeader = (text: string) => text.split('\n')[0].split('|')[1].replaceAll('}', '').trim();
```

Comment:
How did you know what to split/replace here?

Reply:
this is the rich text format of echarts, let me add a comment here